pallets/msa/src/tests.rs

```diff
@@ -490,43 +490,43 @@ pub fn create_sponsored_account_with_delegation_with_valid_input_should_succeed(
 	new_test_ext().execute_with(|| {
 		// arrange
 		let (key_pair, _) = sr25519::Pair::generate();
-		let delegate_account = key_pair.public();
+		let provider_account = key_pair.public();
 
 		let (key_pair_delegator, _) = sr25519::Pair::generate();
 		let delegator_account = key_pair_delegator.public();
 
-		let add_delegate_payload = AddDelegate { authorized_msa_id: 1u64.into(), permission: 0 };
-		let encode_add_delegate_data = wrap_binary_data(add_delegate_payload.encode());
+		let add_provider_payload = AddProvider { authorized_msa_id: 1u64.into(), permission: 0 };
+		let encode_add_provider_data = wrap_binary_data(add_provider_payload.encode());
 
-		let signature: MultiSignature = key_pair_delegator.sign(&encode_add_delegate_data).into();
+		let signature: MultiSignature = key_pair_delegator.sign(&encode_add_provider_data).into();
 
-		assert_ok!(Msa::create(Origin::signed(delegate_account.into())));
+		assert_ok!(Msa::create(Origin::signed(provider_account.into())));
 
 		// act
 		assert_ok!(Msa::create_sponsored_account_with_delegation(
-			Origin::signed(delegate_account.into()),
+			Origin::signed(provider_account.into()),
 			delegator_account.clone().into(),
 			signature.clone(),
-			add_delegate_payload.clone()
+			add_provider_payload.clone()
 		));
 
 		// assert
 		let key_info = Msa::get_key_info(AccountId32::new(delegator_account.0));
 		assert_eq!(key_info.unwrap().msa_id, 2);
 
-		let delegated = Msa::get_delegate_info_of(Delegate(1), Delegator(2));
-		assert_eq!(delegated.is_some(), true);
+		let providerd = Msa::get_provider_info_of(Provider(1), Delegator(2));
```

Comment:
Overeager replacement - I'd rename this to "provider_info"

Reply:
:D good catch hence was this spin off , will take a keen look